src/v/s3/client.cc

```diff
@@ -527,20 +531,31 @@ ss::future<> client_pool::stop() {
 ///         are in use)
 ss::future<client_pool::client_lease> client_pool::acquire() {
     gate_guard guard(_gate);
-    if (_pool.empty()) {
-        if (_policy == client_pool_overdraft_policy::wait_if_empty) {
-            co_await _cvar.wait([this] { return !_pool.empty(); });
-        } else {
-            auto cl = ss::make_shared<client>(_config, _as);
-            _pool.emplace_back(std::move(cl));
+    try {
+        while (_pool.empty() && !_gate.is_closed()) {
+            if (_policy == client_pool_overdraft_policy::wait_if_empty) {
+                co_await _cvar.wait();
+            } else {
+                auto cl = ss::make_shared<client>(_config, _as);
+                _pool.emplace_back(std::move(cl));
+            }
         }
+    } catch (const ss::broken_condition_variable&) {
     }
+    if (_gate.is_closed() || _as.abort_requested()) {
+        throw ss::gate_closed_exception();
```

Comment:
If you've reached this point, you've exited the while loop meaning the pool has something in it, then if the gate is closed or abort is requested you will throw instead of returning the item referencing the newly created item, leaving it in the pool anyway. Is this desired behavior?

Reply:
the desired behavior is to trigger `gate_closed_exception` if `stop` was called previously
we have two cases:
- the stop is called and the acquire is called and we're getting the exception
- the acquire is called and it starts to wait on condition variable, then the 'stop' is called, I want the acquire to throw the same exception